auto_dev/services/publish/index.py

```diff
@@ -0,0 +1,171 @@
+        # we have to do a horrible hack here, regards to the customs as they are not being published.
+        # please see issue.
+        for package in agent_config_yaml[0]["customs"]:
+            custom_id = PublicId.from_str(package)
+            # We need to copy the customs to the parent now.
+            customs_path = Path("vendor") / custom_id.author / "customs" / custom_id.name
+            package_path = self._get_package_path(custom_id)
+            if not package_path.exists():
+                shutil.copytree(
+                    customs_path,
+                    package_path,
+                )
+
+        for command in publish_commands:
+            command = CommandExecutor(
+                command.split(" "),
+            )
+            result = command.execute(verbose=self.verbose)
+            if not result:
+                msg = f"""
+                Command failed: {command.command}
+                Error: {command.stderr}
+                stdout: {command.stdout}"""
+                raise OperationError(msg)
+
+    def publish_agent(
+        self,
+        public_id: Optional[PublicId] = None,
+        lock_type: Optional[LockType] = None,
+        force: bool = False,
+    ) -> None:
+        """Publish an agent.
+
+        Args:
+            public_id: Optional. The public_id of the agent. If not provided,
+                assumes we're already in the agent directory.
+            lock_type: the type of lock to apply (dev or third_party). If provided, packages will be locked.
+            force: If True, remove existing package before publishing.
+
+        Raises:
+            OperationError: if the command fails.
+        """
+        if public_id:
+            logger.info(f"Publishing agent {public_id}")
+            if Path(public_id.name).exists():
+                with change_dir(public_id.name):
+                    self._publish_agent_internal(force)
+            else:
+                raise OperationError(f"Agent directory {public_id.name} does not exist")
+        else:
+            logger.info("Publishing agent from current directory")
+            # Verify we're in an agent directory by checking for aea-config.yaml
+            if not Path("aea-config.yaml").exists():
+                raise OperationError("Not in an agent directory (aea-config.yaml not found)")
+            self._publish_agent_internal(force)
+
+        logger.info("Agent published!")
+        if lock_type is not None:
+            self.lock_packages(lock_type)
+
+    def lock_packages(self, lock_type: LockType = LockType.DEV) -> None:
+        """Lock the packages after publishing.
+
+        Args:
+            lock_type: the type of lock to apply (dev or third_party).
+        """
+
+        logger.info(f"Locking packages as {lock_type.value}")
+
+        command = CommandExecutor(
+            ["bash", "-c", f"yes {lock_type.value} | autonomy packages lock"],
+        )
+        result = command.execute(verbose=self.verbose)
```

Comment:
this is no good as it sets all packages as dev.
I dont think that is desired behaviour.

Reply:
do we want them to enter in the options?

Reply:
i guess this behavior isn't horrible, i think a optional for default setting would be good (example off the top of my head being if we are ejecting you can assume its dev